### ISSUE TITLE
Training on lit-llama failed to get convergence

### ISSUE THREAD

**Issue by hx-Tang** [Liuhong99/Sophia#24]
Hi, @Liuhong99 

I have integrated Sophia on lit-llama and tested its performance on a subset of red-pajama. For some hyper-parameters it can get convergence and achieve similar of better results then adamw. But for larger learning rate, Sophia failed to get convergence. Have you encounter similar situation? Note that the hessian_norm and hessian_norm2 increases rapidly as the loss stop decreasing. Any clue for what's happening?

Here is the code integrating sophia on lit-llama.
https://github.com/hx-Tang/llama-palyground-thx
I made small adjustment on 'bs' used in .step(). As it's a constant, I set the value when initializing sophia.

Here is the wandb report:
https://api.wandb.ai/links/1998thx/e68qw2yg

i'm using the following hyper-parameters for both adamw and sophia
'''
learning_rate = 4e-4
batch_size = 125
micro_batch_size = 8
weight_decay = 1e-1
beta1 = 0.965
beta2 = 0.99
grad_clip = 1.0
min_lr = 4e-5

hess_interval = 10
rho = 0.02
'''

**Comment by vmarkovtsev** [comment]
@hx-Tang would it be hard to check whether the hessian is non-negative when it doesn't converge? I have some suspicions in https://github.com/Liuhong99/Sophia/issues/34

**Comment by Liuhong99** [comment]
I did not have a chance to try lit-llama.  

> For some hyper-parameters it can get convergence and achieve similar of better results then adamw. But for larger learning rate, Sophia failed to get convergence. 

It's better to increase `rho` as you use larger learning rate. In the meantime, you can increase weight decay a bit maybe to 0.2.

>  Note that the hessian_norm and hessian_norm2 increases rapidly as the loss stop decreasing.

Do you know how the win_rate changes in that case?

**Comment by Liuhong99** [comment]
We still don't have time to investigate in lit-llama, but we will be able to play with Sophia on llama models later this year. Thanks!
